docs/source/configuration/overview.mdx

```diff
@@ -615,6 +615,18 @@ Note that the router calculates the recursion depth for each operation and fragm
 
 > In versions of the Apollo Router prior to 1.17, this limit was defined via the config option `experimental_parser_recursion_limit`.
 
+### GraphQL Validation
+
+We are experimenting with a new GraphQL validation implementation written in Rust. The legacy implementation is part of the JavaScript query planner. This is part of a project to remove JavaScript from the Router to improve performance and memory behavior.
+
+To opt in to the new validation implementation, set:
+
+```yaml {4,8} title="router.yaml"
+experimental_graphql_validation: new
```

Comment:
Gonna suggest something here when I get out of meetings

Reply:
Suggest changing it to `experimental_graph_validation_mode` 
Reason being that `experimental_graph_validation` as a top level key makes it sound like a feature that people didn't have before.